src/components/AudioPlayer/AudioPlayer.js

```diff
@@ -0,0 +1,205 @@
+import { useEffect, useRef, useState } from 'react'
+
+import AudioControls from '../AudioControls/AudioControls.js'
+import PropTypes from 'prop-types'
+import styled from 'styled-components/macro'
+
+AudioPlayer.propTyped = {
+  audioObject: PropTypes.object,
+}
+
+export default function AudioPlayer({ audioObject }) {
+  const [progress, setProgress] = useState(0)
+  const [isPlaying, setIsPlaying] = useState(false)
+  const [isReady, setIsReady] = useState(false)
+
+  const { image, title, author, audioSrc } = audioObject
+  const audioRef = useRef(new Audio(audioSrc))
+  const intervalRef = useRef()
+
+  const { duration } = audioRef.current
+
+  // some setup before playing
+  useEffect(() => {
+    audioRef.current = new Audio(audioSrc)
+    setProgress(audioRef.current.currentTime)
+
+    if (isReady) {
+      audioRef.current.play()
+      setIsPlaying(true)
+      startTimer()
+    }
+    if (!isReady) setIsReady(true)
+  }, [isReady, audioSrc])
+
+  // handle changes when clicking play and pause button
+  useEffect(() => {
+    if (isPlaying) {
+      audioRef.current.play()
+      startTimer()
+    }
+    if (!isPlaying) {
+      audioRef.current.pause()
+      clearInterval(intervalRef.current)
+    }
+  }, [isPlaying])
+
+  // cleanup when closing or switching audioSrc
+  useEffect(() => {
+    return () => {
+      audioRef.current.pause()
+      clearInterval(intervalRef.current)
+    }
+  }, [audioSrc])
+
+  return (
+    <AudioContainer>
+      <Progressbar
+        type="range"
+        readOnly
+        value={progress}
+        step="1"
+        min="0"
+        max={duration ? duration : `${duration}`} // duration is NaN at first
+      />
+      <TrackInfo>
+        <Image src={image} alt={`track artork for ${title} by ${author}`} />
+        <TitleAndAuthorContainer>
+          <Title>{title}</Title>
+          <Author>{author}</Author>
+        </TitleAndAuthorContainer>
+      </TrackInfo>
+      <AudioControls isPlaying={isPlaying} onPlayPauseClick={setIsPlaying} />
+    </AudioContainer>
+  )
+
+  function startTimer() {
+    // clear any timers
+    clearInterval(intervalRef.current)
+
+    // Pause when audio has ended and if not has ended update the progress bar
+    intervalRef.current = setInterval(() => {
+      if (audioRef.current.ended) audioRef.current.pause()
+      if (!audioRef.current.ended) setProgress(audioRef.current.currentTime)
+      // this function should run every second
+    }, [1000])
```

Comment:
maybe get rid of the comments your code should explain itself

Reply:
For me it would be hard to understand the code without comments.